official/recommendation/data_async_generation.py

```diff
@@ -436,13 +440,26 @@ def remove_alive_file():
     gc.collect()
 
 
+def _parse_flagfile():
+  """Fill flags with flagfile."""
+  flagfile = os.path.join(flags.FLAGS.data_dir,
+                          rconst.FLAGFILE)
+  tf.logging.info("Waiting for flagfile to appear at {}..."
+                  .format(flagfile))
+  while not tf.gfile.Exists(flagfile):
```

Comment:
You should wait a max of rconst.TIMEOUT_SECONDS here, before giving up and exiting.